pkg/dockerfilegen/generator.go

```diff
@@ -152,9 +152,35 @@ func generateDockerfile(params Params, mainPackagesPaths sets.Set[string]) error
 		goVersion = strings.Join(strings.Split(goVersion, ".")[0:2], ".")
 	}
 
+	metadata, err := project.ReadMetadataFile(params.ProjectFilePath)
+	if err != nil {
+		if !errors.Is(err, os.ErrNotExist) {
+			return fmt.Errorf("%w: Failed to read project metadata file: %w",
+				ErrBadConf, errors.WithStack(err))
+		}
+		log.Println("File not found:", params.ProjectFilePath, "(Using defaults)")
+		metadata = project.DefaultMetadata()
+	}
+
+	rhelVersion := "rhel-9"
+	if metadata.Project.Tag != "" {
+		// tag before knative-v1.17
+		minorVersion, err := strconv.Atoi(strings.Replace(metadata.Project.Tag, "knative-v1.", "", 1))
+		if err != nil {
+			if minorVersion < 17 {
+				rhelVersion = "rhel-8"
+			}
+		}
+	} else {
+		// version before 1.37+
+		if metadata.Project.Version == "1.36.1" || metadata.Project.Version == "1.35.1" {
```

Comment:
Maybe we can use just the safer side if we do a patch release 1.36.2 or newer

`minor, _ := strconv.Atoi(strings.Split(metadata.Project.Version, ".")[1])
if minor <= 36 {
	rhelVersion = "rhel-8"
} else {
	rhelVersion = "rhel-9"
}
`